uaclient/mainwindow.py

```diff
@@ -154,7 +154,13 @@ def _save_state(self):
 
         self._settings.setValue("main_window/geometry", self.saveGeometry())
         self._settings.setValue("main_window/state", self.saveState())
-        self._settings.setValue("main_window/address_list", self._address_list)
+
+        self._settings.beginWriteArray("main_window/address_list")
+        for i in range(len(self._address_list)):
+            self._settings.setArrayIndex(i)
+            self._settings.setValue("address", self._address_list[i])
```

Comment:
`enumerate()` might be cleaner here:

```suggestion
        for index, value in enumerate(self._address_list):
            self._settings.setArrayIndex(index)
            self._settings.setValue("address", value)
```

Thoughts?